### PR TITLE
Problem: we want to use TH1-TH4 in port names still

### PR DESCRIPTION
**Pull request by vyskocilm**
Solution: create compat symlinks to real tty devices + fix compatiblity
with HW rev 00

Signed-off-by: Michal Vyskocil <MichalVyskocil@eaton.com>

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -365,8 +365,13 @@ obs_SCRIPTS	=  obs/preinstallimage-bios.sh
 EXTRA_DIST	+= $(obs_SCRIPTS)
 
 ipc_meta_setupdir = $(datadir)/@PACKAGE@/setup/
-ipc_meta_setup_SCRIPTS = $(top_srcdir)/setup/*.sh
-EXTRA_DIST += $(ipc_meta_setup_SCRIPTS)
+ipc_meta_setup_SCRIPTS = \
+	$(top_srcdir)/setup/20-fty-compat.sh \
+	$(top_srcdir)/setup/20-th-names.sh
```
Comment (jimklimov) [inline Makefile.am]: Generated files appear under `$(top_builddir)`

File: setup/20-th-names.sh.in [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# REVISION 00 with TH1-TH4 as /dev/ttyS13 - /dev/ttyS16
+# others with TH1-TH4 as /dev/ttyS9 - /dev/ttyS12
+# Installed udev rules belongs to newest revisions, so
+# 
+
+# copy&paste from tntnet-ExecStartPre.sh.in
+JSONSH="@datadir@/@PACKAGE@/scripts/JSON.sh"
+get_a_string_arg() { "$JSONSH" -x "$1" | (IFS="`printf '\t'`" read K V; echo "$V") | sed 's,^"\(.*\)",\1,' ; }
+J="/etc/release-details.json"
+R=`get_a_string_arg hardware-spec-revision < $J`
```
Comment (jimklimov) [inline setup/20-th-names.sh.in]: Surround with doublequotes `R="$(...)"`

File: setup/20-th-names.sh.in [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+# we don't need to change rules for revisions != 0
+[ $R = "00" ] || exit 0
```
Comment (jimklimov) [inline setup/20-th-names.sh.in]: Surround with doublequotes `"$R" = "00"`

File: Makefile.am (hunk 1 above)
Comment (vyskocilm) [inline Makefile.am]: done

File: setup/20-th-names.sh.in (hunk 2 above)
Comment (vyskocilm) [inline setup/20-th-names.sh.in]: done

File: setup/20-th-names.sh.in (hunk 3 above)
Comment (vyskocilm) [inline setup/20-th-names.sh.in]: done

**Comment by jimklimov** [comment]
Almost good - the distclean did not wipe the generated `./setup/20-th-names.sh` (viz (DIST)CLEANFILES).

**Pull request merged**